net/Socket.hpp

```diff
@@ -1241,6 +1236,19 @@ class StreamSocket : public Socket,
             --ExternalConnectionCount;
     }
 
+    /// Emit 'onDisconnect' if it has not been done
+    void ensureDisconnected()
+    {
+        if (!_doneDisconnect)
+        {
+            ASSERT_CORRECT_SOCKET_THREAD(this);
```

Comment:
Do we want to do this check unconditionally?

Sockets shouldn't migrate after disconnecting, so should catch violations.

Reply:
Surely we can do - good idea. I expect there are some shutdown/cleanup cases but - we can get the thread affinity issues sorted out there if necessary. Still I'd like to get this in first =)